generate_release_notes.py

```diff
@@ -125,180 +149,258 @@ def get_repo_list(github_org, github):
     return sorted(repo_list)
 
 def get_repo(repo_name, github):
-    msg_failure = repo_name + ": repo does not exist or bad token"
+    """
+    Fetch a repository by name from GitHub.
+
+    Args:
+        repo_name (str): The full name of the repository.
+        github (Github): The authenticated GitHub instance.
+
+    Returns:
+        Repository: A GitHub repository object.
+    """
+    msg_failure = f"{repo_name}: repo does not exist or bad token"
     try:
         repo = github.get_repo(repo_name)
-    except Exception as e:
-        logging.error(e)
-        logging.error(msg_failure)
+    except GithubException as error:
+        logging.error("%s", error)
+        logging.error("%s", msg_failure)
         sys.exit(1)
     return repo
 
-def get_release_notes(name, version, issue_titles_bugs, issue_titles_epics, issue_titles_other, commit_only, pull_requests_missing_epics):
-    release_notes = f"\n\n## {name} - {version}\n"
-    if issue_titles_bugs:
-        release_notes += "\n\n#### Bugs & Anomalies\n"
-        release_notes += "* " + "\n* ".join(sorted(set(issue_titles_bugs)))
+def get_release_notes(name, version, issue_titles_epics,
+                      issue_titles_other, github_issues, pr_missing_jira_issues, commit_only, pr_mapping):
+    """
+    Format release notes for the given repository.
+
+    Args:
+        name (str): The repository name.
+        version (str): The release version.
+        issue_titles_epics (set): Titles of unique Jira epics.
+        issue_titles_other (list): Titles of other issues.
+        github_issues (list): GitHub issues.
+        pr_missing_jira_issues (list): PRs missing Jira items or GitHub issues.
+        commit_only (list): Commits missing issues.
```

Comment:
What is a commit missing an issue? Is that a PR missing an issue? If we merge a PR without an issue and the PR includes multiple commits, will we get a bunch of commits missing issues or a PR missing Jira/Github issue or both?

Reply:
So commit missing issue refers to a commit that was made directly to the branch and not associated with any pull request or linked with Jira/GitHub issues. So If no PR is found for a commit, the commit is added to the commit_only list and these are added under orphaned commits at the end. PR missing an issue refers to when a PR is found but does not have any linked Jira/GitHub issues. we will have some commits missing with issues that are captured under each repo but I think as this will be in our intermediate output file and we just use this to make a final notes.